core/chains/evm/config/config.go

```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/config"
 )
 
+//go:generate mockery --quiet --name EVM --output ./mocks/ --case=underscore
```

Comment:
We have moved away from mocking config interfaces since they were cause a lot of problems. How is this one being used?

Reply:
```
if tx.FeeLimit == 0 {
		tx.FeeLimit = chain.Config().EVM().GasEstimator().LimitDefault()
}
```
https://github.com/smartcontractkit/chainlink/blob/e2dde3bb0e90d3c97eb8ffad3055d6a56443ab93/core/web/evm_transactions_controller.go#L142

This is the only way that I could get access to the default gas limit. 
Is it better to introduce new method on `chain.GasEstimator() ` that would return default gas limit?

Reply:
I don't understand what you mean. Are we able to use a fake config type? Or a real config type populated with the values that we need? Mockery types comes with a ton of extra baggage.

Reply:
For example: https://github.com/smartcontractkit/chainlink/blob/1208fb3d3dec36876dfd74a9a05db437840ce9fc/core/chains/evm/txmgr/test_helpers.go#L43